lms/djangoapps/instructor/enrollment.py

```diff
@@ -286,36 +291,53 @@ def reset_student_attempts(course_id, student, module_state_key, requesting_user
             text_type(module_state_key),
         )
 
-    module_to_reset = StudentModule.objects.get(
-        student_id=student.id,
-        course_id=course_id,
-        module_state_key=module_state_key
-    )
-
-    if delete_module:
-        module_to_reset.delete()
-        create_new_event_transaction_id()
-        set_event_transaction_type(grades_events.STATE_DELETED_EVENT_TYPE)
-        tracker.emit(
-            six.text_type(grades_events.STATE_DELETED_EVENT_TYPE),
-            {
-                'user_id': six.text_type(student.id),
-                'course_id': six.text_type(course_id),
-                'problem_id': six.text_type(module_state_key),
-                'instructor_id': six.text_type(requesting_user.id),
-                'event_transaction_id': six.text_type(get_event_transaction_id()),
-                'event_transaction_type': six.text_type(grades_events.STATE_DELETED_EVENT_TYPE),
-            }
-        )
-        if not submission_cleared:
-            _fire_score_changed_for_block(
-                course_id,
-                student,
-                block,
-                module_state_key,
+    def _reset_or_delete_module(studentmodule):
+        if delete_module:
+            studentmodule.delete()
+            create_new_event_transaction_id()
+            set_event_transaction_type(grades_events.STATE_DELETED_EVENT_TYPE)
+            tracker.emit(
+                six.text_type(grades_events.STATE_DELETED_EVENT_TYPE),
+                {
+                    'user_id': six.text_type(student.id),
+                    'course_id': six.text_type(course_id),
+                    'problem_id': six.text_type(module_state_key),
+                    'instructor_id': six.text_type(requesting_user.id),
+                    'event_transaction_id': six.text_type(get_event_transaction_id()),
+                    'event_transaction_type': six.text_type(grades_events.STATE_DELETED_EVENT_TYPE),
+                }
             )
+            if not submission_cleared:
+                _fire_score_changed_for_block(
+                    course_id,
+                    student,
+                    block,
+                    module_state_key,
+                )
+        else:
+            _reset_module_attempts(studentmodule)
+
+    team = None
+    if teams_enabled:
+        from lms.djangoapps.teams.api import get_team_for_user_course_topic
+        team = get_team_for_user_course_topic(student, str(course_id), selected_teamset_id)
+    if team:
+        modules_to_reset = StudentModule.objects.filter(
```

Comment:
What happens to the user's shared state if they're moved from one team to another (e.g. one team has a few members quit for some reason and so teams are redistributed)?

Reply:
I don't believe anything actually happens to their StudentModule state. If a user leaves a team, they no longer have access to files they uploaded as a part of a team, but their actual student module state will remain untouched. https://openedx.atlassian.net/wiki/spaces/MS/pages/1074004056/Team-enabled+ORAs+MVP (Down under "Decisions Captured from Prior Discovery Work")

Reply:
Okay, so it's quite possible I'm not thinking of this correctly because I'm not that familiar with how teams work in ORA2, but what happens if:

1. Lyla was in Group A and has some state saved there, maybe something was even submitted.
2. Group A fell apart for some reason, and Lyla was moved to Group B. But because that team hasn't submitted yet, the course team resets Lyla's attempts/state with the intention of giving her a clean slate to be part of Group B.

Will this code wipe out the student state (including any saved draft submissions) for all of Group B? If so, is that okay?